src/service/performance-impl.js

```diff
@@ -198,20 +189,10 @@ export class Performance {
    * @private
    */
   whenViewportLayoutComplete_() {
-    return this.whenReadyToRetrieveResources_().then(() => {
-      return Promise.all(this.resources_.getResourcesInViewport().map(r => {
-        return r.loadedOnce();
-      }));
-    });
-  }
-
-  /**
-   * Returns a promise that is resolved when the document is ready and
-   * after a microtask delay.
-   * @return {!Promise}
-   */
-  whenReadyToRetrieveResources_() {
-    return this.whenReadyToRetrieveResourcesPromise_;
+    const rect = layoutRectLtwh(
```

Comment:
In this case it could. But in general, I wanted to avoid creating this signal by default. E.g. without analytics and CSI reporting, this signal is not needed, but it's currently somewhat expensive. WDYT?